crypto/s2n_composite_cipher_aes_sha.c

```diff
@@ -167,17 +167,22 @@ static int s2n_composite_cipher_aes_sha_encrypt(struct s2n_session_key *key, str
     POSIX_ENSURE_EQ(out->size, in->size);
 
     POSIX_GUARD_OSSL(EVP_EncryptInit_ex(key->evp_cipher_ctx, NULL, NULL, NULL, iv->data), S2N_ERR_KEY_INIT);
-    POSIX_GUARD_OSSL(EVP_Cipher(key->evp_cipher_ctx, out->data, in->data, in->size), S2N_ERR_ENCRYPT);
+
+    int len = 0;
+    POSIX_GUARD_OSSL(EVP_EncryptUpdate(key->evp_cipher_ctx, out->data, &len, in->data, in->size), S2N_ERR_ENCRYPT);
+
+    S2N_ERROR_IF(len != in->size, S2N_ERR_ENCRYPT);
```

Comment:
I'm not sure why you combined your PR changes into one PR. This makes it difficult to review a PR because your description of changes doesn't match the PR anymore. Unless you have a good reason I would keep these kinds of changes separate in the future.

Reply:
> This change is based on top of a https://github.com/aws/s2n-tls/pull/3466 and the relevant commit is https://github.com/aws/s2n-tls/commit/a7f134edde30574212b66c488b22f021811a0e29

The idea is to merge the other PR first since its super close. I was testing with this rebased on top of the other to make sure all tests were passing locally.